specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-08-01/extensions.json

```diff
@@ -0,0 +1,707 @@
+            "Creating",
+            "Updating",
+            "Deleting",
+            "Moving",
+            "PartiallySucceeded",
+            "PartiallyConnected",
+            "InProgress",
+            "Accepted",
+            "Provisioning"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          },
+          "readOnly": true
+        },
+        "extensionParameters": {
+          "description": "Parameters specific to this extension type.",
+          "$ref": "#/definitions/ExtensionParameters",
+          "x-ms-client-flatten": true
+        },
+        "aggregateState": {
+          "description": "Aggregate state of Arc Extensions across the nodes in this HCI cluster.",
+          "type": "string",
+          "enum": [
+            "NotSpecified",
+            "Error",
+            "Succeeded",
+            "Canceled",
+            "Failed",
+            "Connected",
+            "Disconnected",
+            "Deleted",
+            "Creating",
+            "Updating",
+            "Deleting",
+            "Moving",
+            "PartiallySucceeded",
+            "PartiallyConnected",
+            "InProgress",
+            "Accepted",
+            "Provisioning",
+            "UpgradeFailedRollbackSucceeded"
+          ],
+          "x-ms-enum": {
+            "name": "ExtensionAggregateState",
+            "modelAsString": true
+          },
+          "readOnly": true
+        },
+        "perNodeExtensionDetails": {
+          "description": "State of Arc Extension in each of the nodes.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PerNodeExtensionState"
+          },
+          "readOnly": true
+        },
+        "managedBy": {
+          "description": "Indicates if the extension is managed by azure or the user.",
+          "type": "string",
+          "enum": [
+            "Azure",
+            "User"
+          ],
+          "x-ms-enum": {
+            "name": "ExtensionManagedBy",
+            "modelAsString": true
+          },
+          "readOnly": true
+        }
+      }
+    },
+    "ExtensionPatchProperties": {
+      "description": "Describes Machine Extension Properties that can be updated.",
+      "type": "object",
+      "properties": {
+        "extensionParameters": {
+          "description": "Describes the properties of a Machine Extension that can be updated.",
+          "$ref": "#/definitions/ExtensionPatchParameters"
+        }
+      }
+    },
+    "ExtensionPatchParameters": {
+      "description": "Describes the properties of a Machine Extension that can be updated.",
+      "type": "object",
+      "properties": {
+        "typeHandlerVersion": {
+          "type": "string",
+          "description": "Specifies the version of the script handler. Latest version would be used if not specified."
+        },
+        "enableAutomaticUpgrade": {
+          "type": "boolean",
```

Comment:
ARM recommends enums over booleans for future proof APIs.

Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.
A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].
Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its cofusing, and no better than a boolean.

Reply:
Hi, this is an existing property and has not been added in this api version.